### ISSUE TITLE
Markdown: Code blocks in numbered lists

### ISSUE THREAD

**Issue by TheJaredWilcurt** [forem/forem#1855]
On GitHub, Reddit, etc. I can do this

1. The first step is to `npm install`
1. Next add this code:
   ```js
   console.log('some code');
   ```
1. Step 3 is to...
1. The the fourth step is....
1. Step 5...

But on dev.to it looks like this:

1. The first step is to `npm install`
1. Next add this code:
```js
console.log('some code');
```
1. Step 3 is to...
1. The the fourth step is....
1. Step 5...

* * *

If there is a way to embed code blocks into ordered lists without breaking the ordering, it is not documented anywhere.

This is also a pattern that is slightly different on many different sites, where the number of spaces required at the start of the code block varies, whether it allows syntax highlighting in list code blocks, whether it allows triple graves, and whether or not the remaining lines need indentation.

There are a lot of possible permutations of how this could be implemented and without a live mardown preview, doing trial-and-error to attempt to make it work is frustrating. You need to add or remove a space at the start of every line of code, click preview, scroll scroll scroll down, and then click preview again to go back and try something else.

**Comment by jessleenyc** [comment]
Thanks for the issue! @maestromac any thoughts on this?

**Comment by aspittel** [comment]
I think this is a bug in RedCarpet which we use for Markdown! Think it would be awesome for someone to tackle that though. https://github.com/vmg/redcarpet/issues/664

**Comment by jessleenyc** [comment]
Good call, going to label this as blocked

**Comment by TheJaredWilcurt** [comment]
The issue you linked to deals with code blocks that have lines that begin with `#` being converted to Headings. My issue is not related to that. It is that **any** codeblock will force the end of the list, rather than being indented inside the list. The code block renders accurately, just not in the correct nesting.

```html
<ol>
  <li>asdf</li>
</ol>
<div class="highlight">
  <pre class="highlight javascript">
    <code>
      ...
    </code>
  </pre>
</div>
<ol>
  <li>qwer</li>
</ol>
```

when it should be doing

```html
<ol>
  <li>asdf
    <div class="highlight">
      <pre class="highlight javascript">
        <code>
          ...
        </code>
      </pre>
    </div>
  </li>
  <li>qwer</li>
</ol>
```

**Comment by TheJaredWilcurt** [comment]
![broken](https://user-images.githubusercontent.com/4629794/58705190-74c8a280-837c-11e9-8840-ed19d3a5aa8b.png)

This issue is still occurring. Again, it is **not** related to the linked issue above. This is a separate issue, and should be fixed as it makes me not want to use this platform.

Until you can solve this issue a work around would be to deliver the markdown code directly to the client and use Marked.js to render the HTML on the fly. Since Marked does not suffer from these bugs.

Or you could just fix this issue in whatever library you are using.

**Comment by jessleenyc** [comment]
Thanks for the additional clarification @TheJaredWilcurt! This issue is approved for anyone who's interested in working on it. Should be able to fix it in the MarkdownParser.

**Comment by bolariin** [comment]
I was able to have code blocks inside numbered lists while retaining the order of the list. It looks the code block shouldn't be on the same line as text within the numbered list. I was able to get the same result from indenting one space, two spaces and four spaces from the position where the text within the numbered list begins.

Using the markdown below, I was able to get the result in image below the markdown text

```
1. The first step is to npm install
2. Next add this code:
    ```
    console.log('some code');
    ```
3. Step 3 is to...
4. The the fourth step is....
5. Step 5...
```

![Code block in numbered lists](https://user-images.githubusercontent.com/24629960/67155150-6fdaa000-f2d7-11e9-953f-3af05b303159.png)

Not sure if it actually an issue but codeblocks can't be the only text within a point in a list. 
This ends up looking like the image below
```
1. The first step is to npm install
2. ```
    console.log('some code');
    ```
3. Step 3 is to...
4. The the fourth step is....
5. Step 5...
``` 

![codeblock alone in ordered list](https://user-images.githubusercontent.com/24629960/67155378-0dd06980-f2dc-11e9-914a-dbef03f30f6e.png)

but on GitHub it actually rendered as expected
1. The first step is to npm install
2. ```
    console.log('some code');
    ```
3. Step 3 is to...
4. The the fourth step is....
5. Step 5...

**Comment by TheJaredWilcurt** [comment]
I can only get it to embed in the list with 4 spaces (an improvement from before), but then it also forces all text content in the rest of the list to be wrapped by a paragraphs.

```html
<ol>
  <li>The first step is to <code>npm install</code></li>
  <li>
    <p>Next add this code:<br></p>
    <pre class="highlight plaintext"><code>console.log('some code');
</code></pre>
  </li>
  <li><p>Step 3 is to...</p></li>
  <li><p>The the fourth step is....</p></li>
  <li><p>Step 5...</p></li>
</ol>
```
which adds a bunch of unwanted padding

**Comment by rhymes** [comment]
@jeikabu in #5350 added this comment:

> Same problem as #1855, but when publishing from RSS where the feed is github pages.
> 
> Understand it's a problem with an upstream package, but for content pulled from elsewhere can the "hack" described in #1855 just be applied? I've been manually fixing up posts for over a year, and doing it with the web-based text editor has gotten... old.

**Comment by rhymes** [comment]
This has been fixed in Redcapet 3.5.0, see https://github.com/vmg/redcarpet/issues/664#issuecomment-787127992

I verified it with the markdown in this comment https://github.com/forem/forem/issues/1855#issuecomment-544223428 - there's a slight UI bug but it's unrelated.

Thus I'm going to close this and surface the bug

**Comment by ViacheslavZyrianov** [comment]
Bug still not fixed
